persist/sqldb/workflow_archive.go

```diff
@@ -174,16 +184,35 @@ func (r *workflowArchive) ListWorkflows(namespace string, name string, namePrefi
 		return nil, err
 	}
 
-	wfs := make(wfv1.Workflows, 0)
-	for _, archivedWf := range archivedWfs {
-		wf := wfv1.Workflow{}
-		err = json.Unmarshal([]byte(archivedWf.Workflow), &wf)
-		if err != nil {
-			log.WithFields(log.Fields{"workflowUID": archivedWf.UID, "workflowName": archivedWf.Name}).Errorln("unable to unmarshal workflow from database")
-		} else {
-			// For backward compatibility, we should label workflow retrieved from DB as Persisted.
-			wf.ObjectMeta.Labels[common.LabelKeyWorkflowArchivingStatus] = "Persisted"
-			wfs = append(wfs, wf)
+	wfs := make(wfv1.Workflows, len(archivedWfs))
+	for i, md := range archivedWfs {
+		labels := make(map[string]string)
+		if err := json.Unmarshal([]byte(md.Labels), &labels); err != nil {
+			return nil, err
+		}
+		// For backward compatibility, we should label workflow retrieved from DB as Persisted.
+		labels[common.LabelKeyWorkflowArchivingStatus] = "Persisted"
+
+		annotations := make(map[string]string)
+		if err := json.Unmarshal([]byte(md.Annotations), &annotations); err != nil {
+			return nil, err
+		}
+
+		wfs[i] = wfv1.Workflow{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              md.Name,
+				Namespace:         md.Namespace,
+				UID:               types.UID(md.UID),
+				CreationTimestamp: v1.Time{Time: md.StartedAt},
+				Labels:            labels,
+				Annotations:       annotations,
+			},
+			Status: wfv1.WorkflowStatus{
+				Phase:      md.Phase,
+				StartedAt:  v1.Time{Time: md.StartedAt},
+				FinishedAt: v1.Time{Time: md.FinishedAt},
+				Progress:   wfv1.Progress(md.Progress),
+			},
```

Comment:
What did you base the metadata requirements on? Based on [`workflows-service.ts`](https://github.com/argoproj/argo-workflows/blob/bbc1c08ce20e15b07de37f61f4a12889ec962623/ui/src/app/shared/services/workflows-service.ts#L39) and [`reports.tsx`](https://github.com/argoproj/argo-workflows/blob/bbc1c08ce20e15b07de37f61f4a12889ec962623/ui/src/app/reports/reports.tsx#L59), we may also want `status.message`, `status.estimatedDuration`, and `spec.suspend`

Reply:
I was checking columns in the workflow UI and decided what metadata to include, but if the fields that you mentioned are required, I can open another PR to add them

Reply:
Technically speaking, `fields` is a query parameter in the API, so ideally we'd dynamically set the fields.

If you didn't notice some of them it may be because you can expand the list?

Reply:
Oh, that could be, I will prepare a PR to add those fields.

Reply:
This seems to be causing a regression now: #13098

Reply:
Hmm looking at https://github.com/argoproj/argo-workflows/pull/11121#discussion_r1469101802 more, it may have also unintentionally introduced a feature where you can get all parts of an Archived Workflows in the `list` API, whereas you could not in 3.4 if I'm reading correctly.

I.e. fixing that bug also removed that feature. I don't necessarily mind that if it wasn't present in 3.4, but I can see how that could cause some back-and-forth turbulence for users (which, well, #11121 caused a lot of regressions unfortunately)

Reply:
PR to add the above missing fields: https://github.com/argoproj/argo-workflows/pull/13136